File: java/parsers/src/main/java/jp/co/teruuu/parser/json/type/JNumber.java
package jp.co.teruuu.parser.json.type;

public class JNumber implements JValue{
    public int value = 0;
    public int decimal = 0;
    public int base = 0;

    public JNumber(int value, int decimal, int base) {
        this.value = value;
        this.decimal = decimal;
        this.base = base;
    }

    @Override
    public boolean equals(Object other) {
        return other instanceof JNumber && this.value == ((JNumber) other).value &&
                this.decimal == ((JNumber) other).decimal &&
                this.base == ((JNumber) other).base;
    }
}

File: scala/parsers/src/main/scala/jp/co/teruuu/parser/json/type/JNumber.scala
package jp.co.teruuu.parser.json.`type`

case class JNumber(value: Int, decimal: Int, base: Int) extends JValue {

  override def equals(obj: Any): Boolean = {
    obj match {
      case obj: JNumber => value == obj.value && decimal == obj.decimal && base == obj.base
      case _ => false
    }
  }

  def toSpray(): spray.json.JsValue = {
    spray.json.JsNumber((value * Math.pow(10, base)))
  }
}
